packages/react/src/components/Modal/Modal.js

```diff
@@ -232,10 +233,10 @@ export default class Modal extends Component {
   startTrap = React.createRef();
   endTrap = React.createRef();
   modalInstanceId = `modal-${getInstanceId()}`;
-  modalLabelId = `${prefix}--modal-header__label--${this.modalInstanceId}`;
-  modalHeadingId = `${prefix}--modal-header__heading--${this.modalInstanceId}`;
-  modalBodyId = `${prefix}--modal-body--${this.modalInstanceId}`;
-  modalCloseButtonClass = `${prefix}--modal-close`;
+  modalLabelId = `${this.context}--modal-header__label--${this.modalInstanceId}`;
+  modalHeadingId = `${this.context}--modal-header__heading--${this.modalInstanceId}`;
+  modalBodyId = `${this.context}--modal-body--${this.modalInstanceId}`;
+  modalCloseButtonClass = `${this.context}--modal-close`;
```

Comment:
I think we'll have to add these to `render()` because `this.context` will not be defined in a `static` class property